swagger.yaml

```diff
@@ -7013,9 +7029,46 @@ definitions:
         description: Indication that the version is the recommended one.
       support_level:
         type: string
-        enum: [beta, production, maintenance]
+        enum: [beta, production, maintenance, end-of-life]
         description: Level of support of the version.
 
+  upgrade-channel:
+    type: object
+    required:
+      - cpu_architecture
+      - channels
+    properties:
+      cpu_architecture:
+        type: string
+        enum: ['x86_64', 'aarch64', 'arm64','ppc64le','s390x','multi']
+        description: The CPU architecture of the image.
+        x-go-custom-tag: gorm:"default:'x86_64'"
+      channels:
+        type: array
+        items:
+          type: string
+          enum: [candidate, fast, stable, eus]
+          description: Release channel.
+
+  release-source:
+    type: object
+    required:
+      - openshift_version
+      - upgrade_channels
+    properties:
+      openshift_version:
+        type: string
+        description: Version of the OpenShift cluster.
+      upgrade_channels:
+        type: array
+        items:
+          $ref: '#/definitions/upgrade-channel'
+
+  release-sources:
+    type: array
+    items:
+      $ref: '#/definitions/release-source'
+
   release-images:
```

Comment:
Super minor but maybe leave this next to `release-image` instead of putting the sources stuff in between the singular and array types.